src/modules/commander/Commander.cpp

```diff
@@ -1872,12 +1872,11 @@ Commander::run()
 		if ((override_auto_mode || override_offboard_mode) && is_rotary_wing
 		    && !in_low_battery_failsafe && !_geofence_warning_action_on) {
 			// transition to previous state if sticks are touched
-			if ((_last_manual_control_setpoint.timestamp != _manual_control_setpoint.timestamp) &&
-			    ((fabsf(_manual_control_setpoint.x - _last_manual_control_setpoint.x) > _min_stick_change) ||
-			     (fabsf(_manual_control_setpoint.y - _last_manual_control_setpoint.y) > _min_stick_change) ||
-			     (fabsf(_manual_control_setpoint.z - _last_manual_control_setpoint.z) > _min_stick_change) ||
-			     (fabsf(_manual_control_setpoint.r - _last_manual_control_setpoint.r) > _min_stick_change))) {
-
+			if (hrt_elapsed_time(&_manual_control_setpoint.timestamp) < 1_s && // don't use uninitialized or old messages
+			    ((fabsf(_manual_control_setpoint.x) > _min_stick_change) ||
+			     (fabsf(_manual_control_setpoint.y) > _min_stick_change) ||
+			     (fabsf(_manual_control_setpoint.z - 0.5f) > _min_stick_change) ||
```

Comment:
This changes the implementation to not look for a change in stick input but a stick deflection from presumably center. I would adjust the description of the threshold but see problems arising from a non-springloaded throttle constantly triggering RC override. Let me quickly test it.

Reply:
How about changing the trigger threshold to something like 80%?